bin/hwctl

```diff
@@ -194,13 +194,15 @@ function storage {
 				exit 1
 			fi
 
-			if [ ! -d "$args" ]; then
-				echo "The specified path ($args) does not exist or is not a directory"
+			mounted_path=$(df -h --output=source,target | grep "${args}\(\s\|$\)" | tr -s ' ' | cut -d' ' -f2-)
```

Comment:
I found an edge case when it comes to Btrfs and subvolumes. A partition may appear for more than one mount.

On a default installation of Fedora Silverblue, there are a lot of mounts but only a few show up (and in reverse order, if that matters - it may actually be in our favor).

```
$ lsblk
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINTS
sr0     11:0    1 1024M  0 rom  
zram0  251:0    0    8G  0 disk [SWAP]
sda    252:0    0   60G  0 disk 
├─sda1 252:1    0    1M  0 part 
├─sda2 252:2    0    1G  0 part /boot
└─sda3 252:3    0   59G  0 part /var/home
                                /var
                                /sysroot/ostree/deploy/fedora/var
                                /usr
                                /etc
                                /
                                /sysroot
$ args=("/dev/sda3"); df -h --output=source,target | grep "${args}\(\s\|$\)" | tr -s ' ' | cut -d' ' -f2-
/sysroot
/var
/var/home
```

On a default Fedora Workstation install, both the root `/` and `/home` mounts are subvolumes on a single partition.

```
$ args=("/dev/nvme0n1p3"); df -h --output=source,target | grep "${args}\(\s\|$\)" | tr -s ' ' | cut -d' ' -f2-
/
/home
```

In both cases, this leads the `! -d $mount_path` check to actually report that it is not valid.

As a quick workaround I was thinking we could add a `tail -n 1` to grab the last mount since the home directory seems to always be listed last on Fedora. It could break down the line, though, and may not work well on Linux distributions. Something we can address in the future.

Reply:
Nice catch. Updated.